### PR TITLE
standardize model factory functions

### PR DESCRIPTION
**Pull request by ziyuan-linn**
Currently, some models do not accept the underlying model parameter. `sentiment` also does not allow parameters to be passed in arbitrary order. This PR standardizes the parameters of the factory functions across all models. 

The standized API:

```js
const myModel = ml5.modelName("underlyingModelName", optionsObject, callback);
```
Each of the three parameters is optional, and they can be in an arbitrary order.

### CODE REVIEW

File: src/HandPose/index.js [hunk 1]
```diff
@@ -58,6 +60,13 @@ class HandPose {
    * @private
    */
   async loadModel() {
+    this.modelName = handleModelName(
+      this.modelName,
+      ["MediaPipeHands"],
+      "MediaPipeHands",
+      "handPose"
+    );
+
```
Comment (lindapaiste) [inline src/HandPose/index.js:69]: Personal preference but I feel like it's weird to set `this.modelName` to one value (the raw string) and then replace it with another value (the validated string).  I would move the `handleModelName` call into the constructor.  That way `this.modelName` is always the validated name and there is no confusion about what value it represents.

**Comment by shiffman** [comment]
Apologies @ziyuan-linn I merged #136 first which introduced some conflicts, can you resolve? Thank you for the comments @lindapaiste!

File: src/HandPose/index.js (hunk 1 above)
Comment (ziyuan-linn) [reply src/HandPose/index.js:69]: Thank you @lindapaiste. This makes sense, I will update it.

**Comment by ziyuan-linn** [comment]
@shiffman This should be good to merge!

**Comment by shiffman** [comment]
Great! Tested examples and merging!

**Pull request merged**